packages/redux-simple-auth/src/middleware.js

```diff
@@ -120,6 +126,35 @@ export default (config = {}) => {
             return response
           })
         }
+        case INVALIDATE_SESSION: {
+          const state = getState()
+          const authenticatorName = getAuthenticator(state)
+
+          if (authenticatorName) {
```

Comment:
I think we should treat this as a failure case and return promise with the dispatched `invalidateSession` action rather than silently failing.

The big question is whether we whether we return a resolved or rejected promise with the result of the dispatched `invalidateSession`. I think it probably makes the most sense to return a rejected promise to signal failure.

That being the case, could we use a guard clause instead of wrapping the entire stuff below? Might ease some of the nesting.

```javascript
if (!authenticatorName) {
  return Promise.reject(dispatch(invalidateSessionFailed()))
}
```

Though one might argue that if you don't have an authenticator name, your session is already invalidated, so this might be the more appropriate implementation.

```javascript
if (!authenticatorName) {
  return Promise.resolve(next(action))
}
```

Thoughts?

Reply:
Kind of torn between the 2 options. I'm leaning towards returning a rejected promise, as getting into a state where you try and invalidate a session that never existed signals to me that there is a problem with the implementers application logic that they should fix.